infrastructure/cdn-in-a-box/ort/traffic_ops_ort/config_files.py

```diff
@@ -60,24 +60,37 @@ class ConfigFile():
 	contents = "" #: The full contents of the file - as configured in TO, not the on-disk contents
 	sanitizedContents = "" #: Will store the contents after sanitization
 
-	def __init__(self, raw:dict = None, toURL:str = "", tsroot:str = "/"):
+	def __init__(self, raw:dict = None, toURL:str = "", tsroot:str = "/", *unused_args, contents: str = None, path: str = None):
 		"""
 		Constructs a :class:`ConfigFile` object from a raw API response
 
 		:param raw: A raw config file from an API response
 		:param toURL: The URL of a valid Traffic Ops host
 		:param tsroot: The absolute path to the root of an Apache Traffic Server installation
+		:param contents: Directly constructs a ConfigFile from the passed contents. Must be used with path, and causes raw to be ignored.
+		:param path: Sets the full path to the file when constructing ConfigFiles directly from contents.
 		:raises ValueError: if ``raw`` does not faithfully represent a configuration file
 
 		>>> a = ConfigFile({"fnameOnDisk": "test",
 		...                 "location": "/path/to",
-		...                 "apiURI":"/test",
-		...                 "scope": servers}, "http://example.com/")
+		...                 "apiUri":"/test",
+		...                 "scope": "servers"}, "http://example.com/")
 		>>> a
 		ConfigFile(path='/path/to/test', URI='http://example.com/test', scope='servers')
 		>>> a.SSLdir
-		"/etc/trafficserver/ssl"
+		'/etc/trafficserver/ssl'
+		>>> ConfigFile(contents='testquest', path='/path/to/test')
+		ConfigFile(path='/path/to/test', URI=None, scope=None)
 		"""
+		self.SSLdir = os.path.join(tsroot, "etc", "trafficserver", "ssl")
```

Comment:
My edge container is getting errors finding the cert. It is looking for `/etc/trafficserver/ssl/video_demo1_mycdn_ciab_test_cert.cer`, but in the container, the file exists at `/opt/trafficserver/etc/trafficserver/ssl/video_demo1_mycdn_ciab_test_cert.cer`. Should the path be prefixed with `tsroot`?:

```
edge_1                      | [Apr 30 19:05:14.221] Server {0x7ff575ebd700} NOTE: loading SSL certificate configuration from /etc/trafficserver/ssl_multicert.config
edge_1                      | [Apr 30 19:05:14.221] Server {0x7ff575ebd700} ERROR: SSL::140692222105344:error:02001002:system library:fopen:No such file or directory:bss_file.c:175:fopen('/etc/trafficserver/ssl/video_demo1_mycdn_ciab_test_cert.cer','r')
edge_1                      | [Apr 30 19:05:14.221] Server {0x7ff575ebd700} ERROR: SSL::140692222105344:error:2006D080:BIO routines:BIO_new_file:no such file:bss_file.c:182
edge_1                      | [Apr 30 19:05:14.221] Server {0x7ff575ebd700} ERROR: failed to load certificate chain from /etc/trafficserver/ssl/video_demo1_mycdn_ciab_test_cert.cer
edge_1                      | [Apr 30 19:05:14.240] Server {0x7ff575ebd700} NOTE: failed to reload ssl_multicert.config
edge_1                      | [Apr 30 19:05:14.240] Server {0x7ff575ebd700} NOTE: loading SSL certificate configuration from /etc/trafficserver/ssl_multicert.config
edge_1                      | [Apr 30 19:05:14.240] Server {0x7ff575ebd700} ERROR: SSL::140692222105344:error:02001002:system library:fopen:No such file or directory:bss_file.c:175:fopen('/etc/trafficserver/ssl/video_demo1_mycdn_ciab_test_cert.cer','r')
edge_1                      | [Apr 30 19:05:14.240] Server {0x7ff575ebd700} ERROR: SSL::140692222105344:error:2006D080:BIO routines:BIO_new_file:no such file:bss_file.c:182
edge_1                      | [Apr 30 19:05:14.241] Server {0x7ff575ebd700} ERROR: failed to load certificate chain from /etc/trafficserver/ssl/video_demo1_mycdn_ciab_test_cert.cer
edge_1                      | [Apr 30 19:05:14.253] Server {0x7ff575ebd700} NOTE: failed to reload ssl_multicert.config
```

Reply:
Actually, that is prefixed with `tsroot`. It's `atstccfg` that's assuming everything's installed under `/opt/trafficserver`. As stated in the Profile's Parameters, that's not where it should go, and is currently a bug in `atstccfg` if it's putting it anywhere else:
```json
		{
			"configFile": "ssl_multicert.config",
			"name": "location",
			"value": "/etc/trafficserver"
		}
```

Reply:
Yeah, I opened an issue for it: https://github.com/apache/trafficcontrol/issues/4677